packages/gatsby/src/utils/__tests__/handle-flags.ts

```diff
@@ -332,6 +330,60 @@ describe(`handle flags`, () => {
       `)
     })
 
+    it(`Display message when LOCKED_IN applies to command different than currently executed`, () => {
+      const response = handleFlags(
+        [
+          {
+            name: `ALWAYS_LOCKED_IN_SET_IN_CONFIG_FOR_DEVELOP`,
+            env: `GATSBY_ALWAYS_LOCKED_IN_SET_IN_CONFIG_FOR_DEVELOP`,
+            command: `develop`,
+            description: `test`,
+            umbrellaIssue: `test`,
+            telemetryId: `test`,
+            experimental: false,
+            // this will always LOCKED IN
+            testFitness: (): fitnessEnum => `LOCKED_IN`,
+          },
+          {
+            name: `SOME_FLAG`,
+            env: `GATSBY_SOME_FLAG`,
+            command: `all`,
+            description: `test`,
+            umbrellaIssue: `test`,
+            telemetryId: `test`,
+            experimental: false,
+            testFitness: (): fitnessEnum => true,
+          },
+        ],
+        {
+          // this has no effect, but we want to show to user that
+          SOME_FLAG: true,
+        },
+        `build`
+      )
+
+      expect(response).toMatchInlineSnapshot(`
+        Object {
+          "enabledConfigFlags": Array [
+            Object {
+              "command": "all",
+              "description": "test",
+              "env": "GATSBY_SOME_FLAG",
+              "experimental": false,
+              "name": "SOME_FLAG",
+              "telemetryId": "test",
+              "testFitness": [Function],
+              "umbrellaIssue": "test",
+            },
+          ],
+          "message": "The following flags are active:
+        - SOME_FLAG · (Umbrella Issue (test)) · test
+        ",
```

Comment:
`message` in this test would be empty string without changes to `handle-flags.ts`